### PR TITLE
Fix #377, Add "OSAL Select" functional tests

### LINKED ISSUE #377: OSAL Select APIs missing functional test
**Is your feature request related to a problem? Please describe.**
OS_SelectMultipe, OS_SelectSingle, OS_SelectFdZero, OS_SelectFdAdd, OS_SelectFdClear, OS_SelectFdIsSet all missing functional tests

**Describe the solution you'd like**
Add tests

**Describe alternatives you've considered**
None

**Additional context**
Certification issue.

**Requester Info**
Jacob Hageman - NASA/GSFC

### PR DESCRIPTION
**Pull request by zanzaben**
**Describe the contribution**
Fixes #377 
Added new functional tests for the the select and select multiple functions. 

**Testing performed**
Ran the new tests. 

**Expected behavior changes**
None

**System(s) tested on**
Ubuntu 20.04

**Contributor Info - All information REQUIRED for consideration of pull request**
Alex Campbell GSFC

### CODE REVIEW

File: src/tests/select-test/select-test.c [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+        StateFlags = OS_STREAM_STATE_WRITABLE;
+        actual = OS_SelectSingle(c_socket_id, &StateFlags, 100);        
+
+        count++;
+    }  
+
+    status = OS_BinSemGive(bin_sem_id);
+
+    if(count >= max_buffer_loop){
+         UtPrintf("Buffer size too large to test Select Single OS_STREAM_STATE_WRITABLE timeout\n");
```
Comment (skliper) [inline src/tests/select-test/select-test.c:394]: Suggest - "Unable to cause OS_STREAM_STATE_WRITEABLE timeout with large looped writes, skipping verification" and mark this as requires analysis.

File: src/tests/select-test/select-test.c [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+        actual   = OS_TimedWrite(c_socket_id, Buf_send_c, sizeof(Buf_send_c), 10);
+
+        StateFlags = OS_STREAM_STATE_WRITABLE;
+        actual = OS_SelectSingle(c_socket_id, &StateFlags, 100);        
+
+        count++;
+    }   
+
+    if(count >= max_buffer_loop){
+         UtPrintf("Buffer size too large to test Select Multiple OS_STREAM_STATE_WRITABLE\n");
```
Comment (skliper) [inline src/tests/select-test/select-test.c:478]: same comment

File: src/tests/select-test/select-test.c [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+osal_id_t     s2_task_id;
+osal_id_t     s_socket_id;
+osal_id_t     s2_socket_id;
+osal_id_t     c_socket_id;
+osal_id_t     c2_socket_id;
+OS_SockAddr_t s_addr;
+OS_SockAddr_t s2_addr;
+OS_SockAddr_t c_addr;
+OS_SockAddr_t c2_addr;
+osal_id_t bin_sem_id;
```
Comment (skliper) [inline src/tests/select-test/select-test.c:47]: Consider a global structure

File: src/tests/select-test/select-test.c [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+int max_buffer_loop = 1000000;
```
Comment (skliper) [inline src/tests/select-test/select-test.c:48]: Consider a `#define`, unless this really is a variable.

File: src/tests/select-test/select-test.c [hunk 5]
```diff
@@ -0,0 +1,539 @@
...
+#include "utassert.h"
+#include "uttest.h"
+#include "utbsp.h"
+
+osal_id_t     s_task_id;
+osal_id_t     s2_task_id;
+osal_id_t     s_socket_id;
+osal_id_t     s2_socket_id;
+osal_id_t     c_socket_id;
+osal_id_t     c2_socket_id;
```
Comment (skliper) [inline src/tests/select-test/select-test.c:42]: A quick description of the socket uses would help clarify, or maybe a more general outline of what this test does in the file description.

File: src/tests/select-test/select-test.c [hunk 6]
```diff
@@ -0,0 +1,539 @@
...
+
+/* *************************************** MAIN ************************************** */
+
+char *           fsAddrPtr = NULL;
+static osal_id_t setup_file(void)
+{
+    osal_id_t id;
+    OS_mkfs(fsAddrPtr, "/ramdev3", "RAM3", 512, 20);
+    OS_mount("/ramdev3", "/drive3");
+    OS_OpenCreate(&id, "/drive3/select_test.txt", OS_FILE_FLAG_CREATE | OS_FILE_FLAG_TRUNCATE, OS_READ_ONLY);
```
Comment (skliper) [inline src/tests/select-test/select-test.c:59]: Provide some context... why "3", general concept, etc.

**Review by skliper — CHANGES_REQUESTED** [review]
Should the client sockets get closed?

**Comment by astrogeco** [comment]
[CCB 2020-12-16](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-December-16-2020) APPROVED

- Run `clang-format` for consistency
- Teardown closes the sockets and are used for all the tests

**Comment by skliper** [comment]
@zanzaben - could you add a more descriptive commit message on the last two commits?  The current text isn't sufficient to describe the change.  Let me know if suggestions would help.

**Pull request merged**